cmd/podman/rmi.go

```diff
@@ -51,28 +52,33 @@ func rmiCmd(c *cli.Context) error {
 		return errors.Errorf("when using the --all switch, you may not pass any images names or IDs")
 	}
 
-	imagesToDelete := args[:]
+	images := args[:]
 	var lastError error
+	var imagesToDelete []*image.Image
 	if removeAll {
-		localImages, err := runtime.GetImages(&libpod.ImageFilterParams{})
+		imagesToDelete, err = runtime.GetImages(&libpod.ImageFilterParams{})
 		if err != nil {
 			return errors.Wrapf(err, "unable to query local images")
 		}
-		for _, image := range localImages {
-			imagesToDelete = append(imagesToDelete, image.ID)
+	} else {
+		// Create image.image objects for deletion from user input
+		for _, i := range images {
+			newImage, err := runtime.ImageRuntime().NewFromLocal(i)
+			if err != nil {
+				return err
```

Comment:
Shouldn't we set lastError here, instead of returning an error? So we proceed if only one or two error?

Reply:
this code is unchanged and is designed how dwalsh wanted it?

Reply:
The way this is, if any image requested are not local, we fail here immediately instead of proceeding. That doesn't seem right?

Reply:
Fixed